tgtg_scanner/notifiers/smtp.py

```diff
@@ -79,29 +96,48 @@ def _stay_connected(self) -> None:
         if status != 250:
             self._connect()
 
-    def _send_mail(self, subject: str, html: str) -> None:
+    def _send_mail(self, subject: str, html: str, item_id: int) -> None:
         """Sends mail with html body"""
         if self.server is None:
             self._connect()
         if self.sender is None or self.recipients is None or self.server is None:
             raise SMTPConfigurationError()
         message = MIMEMultipart("alternative")
         message["From"] = self.sender
-        message["To"] = ", ".join(self.recipients)
+
+        # Contains either the main recipient(s) or recipient(s) that should be
+        # notified for the specific item. First, initalize with main recipient(s)
+        recipients = self.recipients
+
+        # Determine recipient(s) based on item_id
+        if self.item_recipients:
+            item = str(item_id)
+            if item in self.item_recipients:
+                # Find addresses based on item to be notified
+                recipients = []
+                recipients_for_item = self.item_recipients[item]
+                if isinstance(recipients_for_item, list):
+                    for address in recipients_for_item:
+                        recipients.append(address)
+                else:
+                    address = recipients_for_item
+                    recipients.append(address)
```

Comment:
if type `dict[str, list[str]]` is ensured you can simply do

```suggestion
            recipients = self.item_recipients.get(str(item_id), self.recipients)
```
This way you could also delete line 110.

The result is a very pythonic one-liner.